cmd/vclusterctl/cmd/connect.go

```diff
@@ -142,6 +144,16 @@ func (cmd *ConnectCmd) Connect(vclusterName string, command []string) error {
 		return err
 	}
 
+	if cmd.BackgroundProxy && localkubernetes.IsDockerInstalledAndUpAndRunning() {
```

Comment:
Add a check here for `cmd.Server` being empty as otherwise starting a proxy container is unnecessary